layer/projects/execution_planner.py

```diff
@@ -131,77 +99,112 @@ def build_execution_plan(
     return execution_plan
 
 
-def topological_sort_grouping(
-    graph: "DiGraph",
-) -> DefaultDict[int, List[PlanNode]]:
-    _graph = graph.copy()
-    res = defaultdict(list)
-    level = 0
-    while _graph:
-        zero_in_degree = [
-            vertex for vertex, degree in _graph.in_degree() if degree == 0
-        ]
-        res[level] = [
-            PlanNode(vertex[0], vertex[1], _graph.nodes[vertex]["entity"])
-            for vertex in zero_in_degree
-        ]
-        _graph.remove_nodes_from(zero_in_degree)
-        level = level + 1
-    return res
+def check_entity_dependencies(definitions: Sequence[FunctionDefinition]) -> None:
+    _build_directed_acyclic_graph(definitions)
```

Comment:
This feels a bit weird. Not necessary to address in this PR, but maybe for followups

Reply:
Weird in what way?